cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java

```diff
@@ -498,16 +512,22 @@ public void getArtifactPlugin(HttpRequest request, HttpResponder responder,
                                 @PathParam("artifact-version") String artifactVersion,
                                 @PathParam("plugin-type") String pluginType,
                                 @PathParam("plugin-name") String pluginName,
-                                @QueryParam("scope") @DefaultValue("user") String scope)
-    throws NamespaceNotFoundException, BadRequestException, ArtifactNotFoundException {
+                                @QueryParam("scope") @DefaultValue("user") String scope,
+                                @QueryParam("versionRange") String versionRange,
```

Comment:
no, that's the plugin name, not the artifact name. The scope is the parent's scope, and the version is the parent's version. This always confuses people though...

If you keep the use case in mind, it is that somebody has specified the plugin's artifact scope and name, and has given a range for the artifact version. If we don't have a filter for the artifact scope and name, there will be problems where you could get the wrong plugin.